src/uct/cuda/cuda_copy/cuda_copy_ep.c

```diff
@@ -55,22 +55,27 @@ ucs_status_t uct_cuda_copy_init_stream(CUstream *stream)
 }
 
 static UCS_F_ALWAYS_INLINE CUstream *
-uct_cuda_copy_get_stream(uct_cuda_copy_iface_t *iface,
+uct_cuda_copy_get_stream(uct_cuda_copy_per_ctx_rsc_t *ctx_rsc,
                          ucs_memory_type_t src_type, ucs_memory_type_t dst_type)
 {
-    CUstream *stream = NULL;
+    static pthread_mutex_t lock = PTHREAD_MUTEX_INITIALIZER;
+    CUstream *stream            = NULL;
```

Comment:
Is the NULL assignment necessary? Line 70 will always overwrite it.

Reply:
Not needed. Will leave it uninitialized.